tests/device_tests/test_device.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+"""Test file for binary running on devices"""
+import numpy as np
+import os
+import unittest
+from nnlib import NNLib as NNLib
+
+
+class TestDevice(unittest.TestCase):
+    """Base class for Device Test."""
+
+
+    def get_param(self, test_case_path):
+        params = {
+            'library':os.path.join(test_case_path, "lib.so"),
+            'input_npy': os.path.join(test_case_path, "input.npy"),
+            'expected_npy': os.path.join(test_case_path, "expected.npy"),
+        }
+
+        return params
+
+    def get_test_cases(self):
+        input_path = os.environ.get('DEVICE_TEST_INPUT_PATH')
+        test_case_paths = [d for d in os.listdir(input_path) if os.path.isdir(os.path.join(input_path,d))]
```

Comment:
i think it's better to use `pathlib` here (and maybe in the other part of this file)
```
import pathlib
p = pathlib.Path(input_path)
test_case_paths = [d for d in p.iterdir() if d.is_dir()]
```

Reply:
Changed to use `pathlib` 👍